js_hw_7/task_02.js

```diff
@@ -0,0 +1,146 @@
+        return `Type: ${this.type} color: ${this.color} weight: ${this.weight} height: ${this.height} placeOfOrigin: ${this.placeOfOrigin}`
+    };
+
+    get color() {
+            return this._color;
+          };
+
+    set color(newColor) {
+        if (newColor !== 'Красный' && newColor !== 'Черный' && newColor !== 'Белый' && newColor !== 'Синий') throw new Error ('Invalid color');
+        this._color = newColor;
+    };      
+};
+
+class Snake extends Animal {
+    #isPoisonous
+    constructor (type, color, weight, height, placeOfOrigin, isPoisonous) {
+        super (type, color, weight, height, placeOfOrigin);
+        this.#isPoisonous = isPoisonous;
+    };
+    checkPoisonous(){
+       if (!(typeof this.#isPoisonous === 'boolean')) throw new Error ('isPoisonous should be boolean value');
+       return this.#isPoisonous;
+    };
+};
+
+// Task 2.
+// 1. Создайте класс Bird с приватным полем isFlying, отнаследовавшись от Animal
+
+class Bird extends Animal {
+    #isFlying;
+    constructor(type, color, weight, height, placeOfOrigin, isFlying) {
+        super(type, color, weight, height, placeOfOrigin);
+        this.#isFlying = isFlying;
+    }
+};
+
+// 2. Создайте класс CatLike с публичным полем isSafeToPet, отнаследовавшись от Animal
+
+class CatLike extends Animal {    
+    constructor (type, color, weight, height, placeOfOrigin, isSafeToPet) {
+        super(type, color, weight, height, placeOfOrigin);
+        this.isSafeToPet = isSafeToPet;
+    }
+};
+
+// 3. Создайте класс Worker, реализующий следующий интерфейс (набор полей и методов):
+//     class Worker
+//       firstName
+//       lastName
+//       phone
+//       getFullName()
+
+class Worker {
+    constructor (firstName, lastName, phone){
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.phone = phone;
+    };
+    getFullName() {
+        return `${this.firstName} ${this.lastName}`
+    }
+};
+
+// 4. Создайте класс Zoo, реализующий следующий интерфейс:
+//     class Zoo
+//       address
+//       title
+//       ticket price
+//       workers: []
+//       animals: [],
+// 5. Добавьте геттеры и сеттеры к полям address, title, ticket price
+// 
+// 6. Добавьте метод addWorker(worker), добавляющий работника в массив workers. 
+//     На вход метод должен принимать объект класса Worker. 
+//     Если объект не является инстансом класса Worker - выкинуть ошибку
+
+// 7. Добавьте метод addAnimal(animal), добавляющий животное в массив animals.
+//     На вход метод должен принимать объект класса Animal, как и любого из его наследников. 
+//     Если объект не является инстансом класса Animal - выкинуть ошибку
+//     ТАКЖЕ, если объект является инстансом класса Snake - выкинуть ошибку с тексом "There will be no snakes, mister Potter!"
+
+// 8. Добавьте методы removeWorker() и removeAnimal() // Подумайте, как будем удалять, по какому полю будем выбирать:)
+
+class Zoo {
+    constructor (address, title, ticketPrice) {
```

Comment:
animals and workers can be probably added in the constructor, so maybe it's ok to handle it's arrays in constructor too)

Reply:
Implemented such way:
class Zoo {
    constructor (address, title, ticketPrice, workers = [], animals = []) {
        this.address = address;
        this.title = title;
        this.ticketPrice = ticketPrice;
        this.workers = workers;        
        this.animals = animals;        
    };